ice-rest-catalog/src/main/java/com/altinity/ice/rest/catalog/internal/rest/RESTCatalogServlet.java

```diff
@@ -64,6 +53,7 @@ public class RESTCatalogServlet extends HttpServlet {
           .put(NoSuchTableException.class, 404)
           .put(NoSuchViewException.class, 404)
           .put(NoSuchIcebergTableException.class, 404)
+          .put(NotFoundException.class, 404)
```

Comment:
I'm afraid we cannot do that. 
Returning 404 on loadTable when table exists but there is an issue with some of the table files feels wrong and is against the the spec https://github.com/apache/iceberg/blob/8da07dcae8ccf5ce1a0c61a7456413c1ce3b65fd/open-api/rest-catalog-open-api.yaml#L976.

Reply:
Yes, you’re right - the table technically exists in the catalog, so returning a `404` would be incorrect.

Unfortunately, `5xx` errors are automatically retried by REST Catalog clients, and we have limited control over that behaviour. One potential workaround could have been using the `Retry-After` header, but I checked, and it has no effect on existing clients (ClickHouse or `ice`). We could modify the clients to respect it, though I’m not sure we want to take that approach yet.

For now, `5xx` responses are retried about 20 times, causing commands to appear to hang.

Let me know if you have other suggestions. 
For now, I’ve removed the `404` mapping, and the issue remains blocked by this behavior.

Reply:
I think 409 could be the compromise we seek (although it would still be not spec compliant + normally it shouldn't be used for something like this). I suggest we merge changes as is as they are useful even without this particular bit and revisit status code in a separate PR. Thank you for the contribution!